File: plot_menu_function.py
from PyQt6.QtWidgets import QMessageBox

def plot_menu_function(player):
    # Simulate toggling settings
    player['setting']['plot_selected'] = not player['setting'].get('plot_selected', False)
    player['setting']['plot_timediagram'] = not player['setting'].get('plot_timediagram', False)
    player['setting']['plot_check_extendline'] = not player['setting'].get('plot_check_extendline', False)
    player['setting']['plot_only_blue_table'] = not player['setting'].get('plot_only_blue_table', False)

    # Ball settings
    color = ['white', 'yellow', 'red']
    tags = ['ball', 'line', 'initialpos', 'marker', 'ghostball', 'diamondpos']

    for bi in range(3):
        player['setting']['ball'][bi] = {}
        for tag in tags:
            player['setting']['ball'][bi][tag] = True  # Simulate toggling

    QMessageBox.information(None, "Plot Menu", "Plot settings updated.")

    # Simulate calling PlayerFunction
    player_function('replot', player)

File: plot_menu_function.m
function plot_menu_function(h, event)

global player param

plot_only_blue_table_old = get(findobj('Tag','plot_check_bluetable'),'Checked');

% change the tick
if ~strcmp(class(h),'double')
   if h.Checked
      h.Checked = 'off';
   else
      h.Checked = 'on';
   end
end


% Plot the selected shots
player.setting.plot_selected = get(findobj('Tag','plot_selected'),'Checked');

% Plot the velocity
player.setting.plot_timediagram = get(findobj('Tag','plot_timediagram'),'Checked');

% Plot extended Shotline of B1
player.setting.plot_check_extendline = get(findobj('Tag','plot_check_extendline'),'Checked');

% Plot table only blue
player.setting.plot_only_blue_table = get(findobj('Tag','plot_check_bluetable'),'Checked');

%% Ball Settings
color = {'white','yellow','red'};
tags = {'ball', 'line', 'initialpos', 'marker', 'ghostball', 'diamondpos'};

for bi = 1:3
   %% bi Ball
   for tagi = 1:length(tags)
      player.setting.ball(bi).(tags{tagi}) = get(findobj('Tag', ['plot_check_',color{bi},'_',tags{tagi}]),'Checked');
   
   end
end

if player.setting.plot_only_blue_table ~= plot_only_blue_table_old
    close_table_figure
end

% Plot selected things
PlayerFunction([],[])
%PlayerFunction('replot',[])
